### PR TITLE
Update requirement to include numpy 2

### PR DESCRIPTION
**Pull request by Ericgig**
Last week we added numpy 2 support but did not update the requirements in `setup.cfg`.

This update dependency to build using numpy 2 instead of oldest-supported-numpy.
I also increased the minimum requirements for matplotlib and scipy to version tested.

I added more test using numpy 2, only one tests build and run using numpy 1.X.
I added more windows / mac tests. Using these os for testing numpy versions and no cython.
`cvxopt` (`dnorm`) does not run with numpy 2 yet, so it's only tested with older numpy version.

Fix sqrtm needing copy. This mean that the test intended to run with numpy 2 previouly did not...

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,8 +7,8 @@ requires = [
     "cython>=0.29.20,<3.0.0; python_version<='3.9'",
     # See https://numpy.org/doc/stable/user/depending_on_numpy.html for
     # the recommended way to build against numpy's C API:
-    "oldest-supported-numpy",
-    "scipy>=1.8",
+    "numpy>=2.0.0",
```
Comment (hodgestar) [inline pyproject.toml:10]: Is building with the latest numpy2 the correct thing to do here? Or should we build with an old numpy2 like with oldest-supported-numpy for numpy 1.x?

**Review by hodgestar — APPROVED** [review]
Looks okay to me. I left one question about whether building with the latest numpy2 is the correct approach now.

File: pyproject.toml (hunk 1 above)
Comment (Ericgig) [reply pyproject.toml:10]: This is how I understand their migration guide. There is no `oldest-supported-numpy2` and that package has not been updated for the release.

**Comment by rgommers** [comment]
Will the NumPy 2.0 changes land in QuTip 5.1.0, or some other version?

File: pyproject.toml (hunk 1 above)
Comment (hodgestar) [reply pyproject.toml:10]: Thanks @rgommers !

**Comment by hodgestar** [comment]
@rgommers I believe @Ericgig is planning to release them in 5.0.3.

**Comment by rgommers** [comment]
Great, thanks!

**Pull request merged**